test/e2e/main_test.go

```diff
@@ -1,11 +1,49 @@
 package e2e
 
 import (
+	"k8s.io/client-go/kubernetes"
 	"testing"
+	"time"
 
+	"github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachineconfig/tracker"
 	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"k8s.io/api/core/v1"
 )
 
+// testContext holds the information related to the entire test suite. This data structure
+// should be instantiated by every test suite, so that we can update the test context to be
+// passed around to get the information which was created within the test suite. For example,
+// if the create test suite creates a Windows Node object, the node object and other related
+// information should be easily accessible by other methods within the same test suite.
+// Some of the fields we have here can be exposed by via flags
+// to the test suite.
+type testContext struct {
+	// numberOfNodes to be used for the test suite
+	numberOfNodes int
+	// osdkTestCtx holds the information related to operator sdk framework's test Context
+	osdkTestCtx *framework.TestCtx
+	// nodes holds the list of nodes
+	nodes []v1.Node
+	// credentials to be used to the node.
+	credentials      []tracker.Credentials
+	retryInterval    time.Duration
+	timeout          time.Duration
+	kubeclient       kubernetes.Interface
+	trackerConfigMap *v1.ConfigMap
+	//TODO: Add deletionCount as well.
+}
+
+// NewTestContext returns a new test context to be used by every test.
+func NewTestContext(t *testing.T, nodeCount int) *testContext {
+	// number of nodes, retry interval and timeout should come from user-input flags
+	return &testContext{numberOfNodes: nodeCount, osdkTestCtx: framework.NewTestCtx(t), retryInterval: time.Second * 5,
+		timeout: time.Minute * 15, kubeclient: framework.Global.KubeClient, nodes: make([]v1.Node, 0, 1)}
+}
+
+func (t *testContext) cleanup() {
+	defer t.osdkTestCtx.Cleanup()
```

Comment:
Why is this a `defer`? Shouldn't that happen at the call site? With this you are taking away the capability is the user does not want to `defer`.

Reply:
Removed it.